compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1314,35 +1305,20 @@ impl<'a> Parser<'a> {
             UnaryFixity::Post => (base.span.shrink_to_lo(), op_span),
         };
 
+        let Ok(base_src) = self.span_to_snippet(base.span)
+        else { return help_base_case(err, base) };
```

Comment:
There's a codepath now where we check for access to the snippet when it isn't strictly necessary for `Standalone`, but that only comes into play in macros and foreign code, this should be fine.